lib/pbench/cli/agent/commands/results/__init__.py

```diff
@@ -1,94 +1,183 @@
-            sys.exit(1)
+        runs_copied = 0
+        failures = 0
 
-        runs_copied += 1
+        for dirent in config.pbench_run.iterdir():
+            if not dirent.is_dir():
+                continue
+            if dirent.name.startswith("tools-") or dirent.name == "tmp":
+                continue
+            result_dir = dirent
+            mrt = MakeResultTb(
+                result_dir,
+                temp_dir,
+                self.context.user,
+                self.context.prefix,
+                self.config,
+                logger,
+            )
 
-    if runs_copied + failures > 0:
-        logger.debug(
-            "successfully moved %s runs, encountered %s failures", runs_copied, failures
-        )
+            try:
+                make_return_val = mrt.check_result_target_dir(result_dir, temp_dir)
+            except FileNotFoundError as e:
+                logger.error(e)
+
+            if make_return_val:
+                failures += 1
+                continue
+            status, result_tb_name = mrt.make_result_tb()
+
+            if status == 0:
+                if os.path.exists(result_tb_name):
+                    assert (
+                        result_tb_name
+                    ), "Logic bomb!  make_result_tb() always returns a tar ball name"
+
+                    crt = CopyResultTb(
+                        self.context.controller, result_tb_name, self.config, logger
+                    )
+                    copy_return_val = crt.check_tb_tbmd5()
+                    if copy_return_val:
+                        failures += 1
+                        continue
+                    ret = crt.copy_result_tb(self.context.token)
+                    if ret:
+                        failures += 1
+                        continue
+            else:
+                logger.error("Error occured while creating Result Tarball")
+                failures += 1
+                continue
+
+            try:
+                # We always remove the constructed tar ball, regardless of success
+                # or failure, since we keep the result directory below on failure.
+                os.remove(result_tb_name)
+                os.remove(f"{result_tb_name}.md5")
+            except OSError:
+                logger.error("rm failed to remove %s and its .md5 file", result_tb_name)
+                failures += 1
+                continue
+
+            try:
+                shutil.rmtree(result_dir)
+            except OSError:
+                logger.error(
+                    "rm failed to remove the %s directory hierarchy", result_dir
+                )
+                failures += 1
+                continue
+
+            runs_copied += 1
+
+        if runs_copied + failures > 0:
+            logger.debug(
+                "successfully moved %s runs, encountered %s failures",
+                runs_copied,
+                failures,
+            )
+            return 1
```

Comment:
I don't think that this code does what we want it to:  on any success (i.e., `runs_copied > 0`) the command will return an error status, regardless of the value of `failures`.

And, the converse is questionable as well:  if both `runs_completed` and `failures` are zero, the command will return "success"....

Reply:
yeah. it isn't appropriate in any case. I think I will just add a status that will include all details and return 1 only if there are failures.

Reply:
Done